File: librarymanager/librarymanager-be/src/main/java/ru/spring5/elena/librarymanager/domain/repository/GenreRepository.java
package ru.spring5.elena.librarymanager.domain.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import ru.spring5.elena.librarymanager.domain.entity.GenreEntity;

@Repository
@Transactional(propagation = Propagation.MANDATORY)
public interface GenreRepository extends JpaRepository<GenreEntity, Long> {

}

File: librarymanager-be/src/main/kotlin/ru/elenacherev/librarymanager/domain/repository/GenreRepository.kt
package ru.elenacherev.librarymanager.domain.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import ru.elenacherev.librarymanager.domain.entity.GenreEntity

@Repository
@Transactional(propagation = Propagation.MANDATORY)
interface GenreRepository : JpaRepository<GenreEntity, Long> 